src/Shared/LoadedType.cs

```diff
@@ -36,31 +38,47 @@ internal LoadedType(Type type, AssemblyLoadInfo assemblyLoadInfo, Assembly loade
             ErrorUtilities.VerifyThrow(type != null, "We must have the type.");
             ErrorUtilities.VerifyThrow(assemblyLoadInfo != null, "We must have the assembly the type was loaded from.");
 
-            _type = type;
+            try
+            {
+                Type t = Type.GetType(type.AssemblyQualifiedName);
+                if (t.Assembly.Location.Equals(loadedAssembly.Location, StringComparison.OrdinalIgnoreCase))
+                {
+                    _type = t;
+                }
+            }
+            catch (Exception) { }
+            _type ??= type;
             _assembly = assemblyLoadInfo;
             _loadedAssembly = loadedAssembly;
 
             CheckForHardcodedSTARequirement();
-            HasLoadInSeparateAppDomainAttribute();
-            HasSTAThreadAttribute();
+            _hasLoadInSeparateAppDomainAttribute = loadedAssembly is null ?
+                this.Type.GetTypeInfo().IsDefined(typeof(LoadInSeparateAppDomainAttribute), true /* inherited */) :
+#if NET35
+                false; // Should never reach here.
+#else
+                CustomAttributeData.GetCustomAttributes(type).Any(attr => attr.AttributeType.Name.Equals("LoadInSeparateAppDomainAttribute"));
```

Comment:
I'd verified that it built correctly but not that the LoadInSeparateAppDomainAttribute was applied. It appears that it was not applied correctly. What do you recommend for getting around that?

Reply:
This is what PathAssemblyResolver is supposed to get around if you give it all the relevant assemblies, right? I'd initially had MSBuild in that list for cases like this, and it failed some test(s)—I forget why exactly—so I took it out. Maybe the solution is to add it back but with some condition?

Reply:
Looking at it again, I had taken out those paths because it failed with an error about not loading the same assembly into a MetadataLoadContext twice. I put in a messy fix:
```C#
string[] runtimePaths = Directory.GetFiles(RuntimeEnvironment.GetRuntimeDirectory(), "*.dll");
            string[] msbuildPaths = Directory.GetFiles(Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location), "*.dll");
            List<string> localPaths = new(Directory.GetFiles(Path.GetDirectoryName(path), "*.dll").Where(localPath => !msbuildPaths.Any(msbuildPath => Path.GetFileName(msbuildPath).Equals(Path.GetFileName(localPath), StringComparison.OrdinalIgnoreCase))));
            localPaths.AddRange(msbuildPaths);
            localPaths.AddRange(runtimePaths.Where(runPath => !msbuildPaths.Any(msbuildPath => Path.GetFileName(msbuildPath).Equals(Path.GetFileName(runPath), StringComparison.OrdinalIgnoreCase))));
```

That meant it was able to load the assembly, but it didn't help with getting LoadInSeparateAppDomainAttribute properly recognized. (I don't think it should matter which LoadInSeparateAppDomainAttribute we use—that is, from which MSBuild assembly—because I was just comparing based on name.)